server/src/main/java/org/opensearch/search/SearchHit.java

```diff
@@ -687,12 +678,21 @@ public XContentBuilder toInnerXContent(XContentBuilder builder, Params params) t
             builder.endObject();
         }
         sortValues.toXContent(builder, params);
-        if (matchedQueries.length > 0) {
-            builder.startArray(Fields.MATCHED_QUERIES);
-            for (String matchedFilter : matchedQueries) {
-                builder.value(matchedFilter);
+        if (matchedQueries != null && !matchedQueries.isEmpty()) {
```

Comment:
The `matchedQueries != null` look redundant, when it could be null?

Reply:
not really, will remove ambiguity

Reply:
fixed